src/library_pthread.js

```diff
@@ -1044,19 +1044,38 @@ var LibraryPThread = {
     return {{{ makeDynCall('ii', 'ptr') }}}(arg);
   },
 
+  // Thread-local guard variable for one-time init of the JS state.
+  $initializedJS: false,
+
+  // A list of queues that receive work before thread initialization.
+  $pendingNotifiedProxyingQueues: [],
```

Comment:
I think its probably makes more sense to do all this queuing and processing in worker.js..